concordium-android-sdk-examples/app/src/main/java/com/example/android_sdk_example/activities/RecoverIdentityActivity.kt

```diff
@@ -0,0 +1,128 @@
+import com.example.android_sdk_example.services.ConcordiumWalletProxyService
+import com.example.android_sdk_example.services.IdentityFetcherService
+import com.example.android_sdk_example.services.wallet_proxy.IdentityProvider
+import com.example.android_sdk_example.ui.Container
+import com.example.android_sdk_example.ui.Menu
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class RecoverIdentityActivity : ComponentActivity() {
+    private fun getRecoverUrl(provider: IdentityProvider, request: String): String {
+        val baseUrl = provider.metadata.recoveryStart
+        return Uri.parse(baseUrl!!).buildUpon().appendQueryParameter("state", request).build()
+            .toString()
+    }
+
+    private fun recoverIdentity(
+        provider: IdentityProvider,
+        global: CryptographicParameters,
+        storage: Storage
+    ) {
+        val request = Requests.createRecoveryRequest(storage.getWallet(), provider, global)
+        val url = getRecoverUrl(provider, request)
+        val identity = IdentityFetcherService().getFromRecovery(url)
+
+        storage.identityProviderIndex.set(provider.ipInfo.ipIdentity.toString())
+        storage.identity.set(jacksonObjectMapper().writeValueAsString(identity))
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val storage = Storage(applicationContext)
+        val global =
+            ConcordiumClientService.getClient().getCryptographicParameters(BlockQuery.BEST)
+
+        setContent {
+            RecoverIdentityView(onSubmit = { provider ->
+                CoroutineScope(Dispatchers.Default).launch {
+                    try {
+                        recoverIdentity(
+                            provider,
+                            global,
+                            storage
+                        )
+                        goToNext(storage, provider, global)
+                    } catch (e: Exception) {
+                        println(e.message)
+                        withContext(Dispatchers.Main) {
+                            Toast.makeText(applicationContext, e.message, Toast.LENGTH_LONG).show()
+                        }
+                    }
+                }
+            })
+        }
+    }
+
+    private fun goToNext(storage: Storage, provider: IdentityProvider, global: CryptographicParameters) {
+        val providerIndex = provider.ipInfo.ipIdentity.value
+        val wallet = storage.getWallet()
+        val credId = wallet.getCredentialId(providerIndex, Constants.IDENTITY_INDEX, Constants.CREDENTIAL_COUNTER, global.onChainCommitmentKey.toHex())
+
+        // Try to get the accountInfo for the identity's account. If successful, jump to the account page, otherwise go to identity page.
```

Comment:
```suggestion
        // The recovered identity might already have an associated account. Check if an account info is available, and if so go directly to the account page. Otherwise go to the identity page.
```